### PR TITLE
added testing of plan functions, small bugfix

### PR DESCRIPTION
**Pull request by midokura-silvia**
Small bugfix in function Plan.replace_worker_ids to support integer ids
Added tests

### CODE REVIEW

File: syft/federated/plan.py [hunk 1]
```diff
@@ -254,12 +254,21 @@ def replace_worker_ids(self, from_worker_id: Union[str, int], to_worker_id: Unio
...
         """
-        for from_id, to_id in [
-            (from_worker_id, to_worker_id),
-            (from_worker_id.encode(), to_worker_id.encode()),
-        ]:
+
+        id_pairs = [(from_worker_id, to_worker_id)]
+        if type(from_worker_id) == str:
+            to_worker_id_encoded = (
+                to_worker_id.encode() if type(to_worker_id) == str else to_worker_id
```
Comment (mari-linhares) [inline syft/federated/plan.py]: Can you cover all 4 possibilities when dealing with strings?

from_worker, to_worker
from_worker, to_worker.encode()
from_worker.encode(), to_worker
from_worker.encode(), to_worker.encode()

File: test/federated/test_plan.py [hunk 2]
```diff
@@ -227,4 +224,97 @@ def my_plan(data):
 
     assert (plan_res == th.tensor([-42, 24, 46])).all()
 
+    # delete remote object before websocket connection termination
+    del x_ptr
+
     server.terminate()
+
+
+def test_replace_worker_ids(hook):
```
Comment (mari-linhares) [inline test/federated/test_plan.py]: There's a lot going on here. Break this into multiple tests and add some comments about what each test is covering.

File: test/federated/test_plan.py [hunk 3]
```diff
@@ -227,4 +224,97 @@ def my_plan(data):
...
+        obj=messages, change_id=10, to_id=100, from_worker="me", to_worker="another"
+    )
+
+    # note that tuples are converted to lists
+    expected = [100, ["worker", "another"], "you", 20, 100, b"you", [30, ["you", "another", "bla"]]]
+
+    assert replaced == expected
+
+
+def test___call__(hook):
```
Comment (mari-linhares) [inline test/federated/test_plan.py]: Break this into two tests: test__call__fn, test__call__method

**Comment by mari-linhares** [comment]
Thanks for this! The tests are super helpful!!!

I've fixed this issue at the train_config branch as well, if you want to have a look.

**Comment by mari-linhares** [comment]
Related to #1989

File: syft/federated/plan.py (hunk 1 above)
Comment (midokura-silvia) [reply syft/federated/plan.py]: We don't need all four possibilities. The outdated code in dev contained the four possibilities. Then the current dev code changed it to (from_worker, to_worker) and (from_worker.encode(), to_worker.encode()) but without taking care of the integers that don't have a encode() function.

As we are replacing, it doesn't make sense to replace from_worker twice. Therefore only two outcomes, not four.

Comment (mari-linhares) [reply syft/federated/plan.py]: Hmm, I would say we need the 4 possibilities (as in train-config) to cover all possible cases, don't we?

My question then is: can we guarantee that with only these two cases all the other cases will work?

Comment (midokura-silvia) [reply syft/federated/plan.py]: Let's think of an example doing string replacements:
"The quick brown b'brown' fox jumps over the lazy b'brown' dog."
If we want to replace brown by black, we would call the replace method with from_worker=brown and to_worker=black.
First call would replace all brown instances with black:
"The quick black b'brown' fox jumps over the lazy b'brown' dog."
The second call would replace all b'brown' instances with b'black':
"The quick black b'black' fox jumps over the lazy b'black' dog."
And we are done.

**Pull request merged**